ember-core/shared/src/test/scala/org/http4s/ember/core/ParserSuite.scala

```diff
@@ -507,6 +524,37 @@ class ParsingSuite extends Http4sSuite {
     } yield body1 == "hello" && body2 == "world").assert
   }
 
+  test(
+    "Parser.Response.parser should parse two responses in a row, even if their reason phrase is missing"
+  ) {
+    val defaultMaxHeaderLength = 4096
+    val respS =
+      Stream(
+        "HTTP/1.1 200\r\n",
+        "Content-Type: text/plain\r\n",
+        "Content-Length: 5\r\n\r\n",
+        "helloHTTP/1.1 200\r\n", // this is the crucial part
```

Comment:
According to the https://datatracker.ietf.org/doc/html/rfc7230#section-3.1.2
```
status-line = HTTP-version SP status-code SP reason-phrase CRLF
```
If I'm not wrong, the whitespace char placed before the reason phrase is mandatory to be. But the reason phrase indeed may be empty, it's a good catch!

Reply:
Damn you're right, this SP is mandatory.
So Ember is correctly parsing status lines, there is no issue here, but I think some servers might remove this SP in their HTTP response since there is nothing to separate. Having something less strict on this point might be a good thing WDYT ?

Reply:
> I think some servers might remove this SP in their HTTP response since there is nothing to separate

That's a very controversial thing, to be honest. From my perspective, we should align with the spec when things are defined univocally. But maybe other folks over http4s have another opinion.

Reply:
On the run at the moment, but it's [Postel's Law](https://en.wikipedia.org/wiki/Robustness_principle), for it's pros and cons.

Questions I consider when we diverge:
- Is there a security implication to relaxing it?
- Do other major implementations accept it?  i.e., are we converging toward a de facto spec?

Reply:
Let me be the responder to Ross's questions.

> Is there a security implication to relaxing it?

Definitely, it isn't.

> Do other major implementations accept it?

I neither have a counterexample of this behaviour.  Have seen in the wild only strict behaviour.

> are we converging toward a de facto spec?

Our implementation is strict, but I am confident it's aligned with the spec.

Reply:
There are far more seasoned folks to speak about that, but it generally reminds me of the question of [parsing filenames in Multipart](https://github.com/http4s/http4s/issues/5809) where we also have strict behaviour.

Reply:
> * Do other major implementations accept it?  i.e., are we converging toward a de facto spec?

@jcchevalier-ledger the one's I'd be interested in are if it works with the [JDK http client](https://github.com/http4s/http4s-jdk-http-client) and if it works with [Netty](https://github.com/http4s/http4s-netty). If neither of those implementations accepts this, then it seems like a bug report for the server :)

Reply:
Hello guys !
Well we started to have this error when we switched the HTTP client of our application from [sttp fs2 client](https://mvnrepository.com/artifact/com.softwaremill.sttp.client3/async-http-client-backend-fs2_3/3.8.15) to ember, so I guess the sttp fs2 (which if I'm not mistaken relies on Netty) handles this non-compliant behavior.